### PR TITLE
feat(perf): Add tags page preliminary UI

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This adds a tags page to add new ways for users to interact with their tags. Now that the backend changes are in, this has been switched to use the new endpoint for histograms and pass params to the existing segment explorer query.

**Note:** Behind a feature flag, this is the first iteration. We're can iterate on the functionality and flow, then I will come back to do a second pass and clean up the edges of the components and add tests.

### CODE REVIEW

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    const newColumns = [...COLUMN_ORDER].map(c => {
+      const newColumn = {...c};
+      if (c.key === 'tagValue') {
+        newColumn.name = tagKey;
+      }
+      return newColumn;
+    });
+
+    return (
+      <SegmentExplorerQuery
```
[the remark this answers is not in the document]
Comment (Zylphrex) [reply static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:318]: Haven't looked at the endpoint yet but I assume this `SegmentExplorerQuery` is unavoidable? But it looks like it only queries on a single tag key.

File: static/app/components/charts/series/heatMapSeries.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import {SeriesDataUnit} from 'app/types/echarts';
+
+export default function HeatMapSeries(
+  props: Omit<EChartOption.SeriesHeatmap, 'data'> & {
+    data?: SeriesDataUnit[] | EChartOption.SeriesHeatmap['data'];
+  } = {}
+): EChartOption.SeriesHeatmap {
+  const {data, ...rest} = props;
+  return {
+    data: data as EChartOption.SeriesHeatmap['data'],
```
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/components/charts/series/heatMapSeries.tsx:15]: Yeah good call, we can probably clean this up. Heatmap data doesn't play well with our existing data format assumptions, but we can fix it over time.

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:318]: It's what populates the data for the table for the tag values for just one key, to match the histogram, but isn't binned. I think theoretically we could use subselects and make 1 query, but for experimentation I'd rather keep it simple.

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    name: '',
+    width: -1,
+    column: {
+      kind: 'field',
+    },
+  },
+];
+
+type Props = {
+  isLoading?: boolean;
```
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:102]: Thanks, was a leftover.

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+    }
+
+    if (column.key === 'frequency') {
+      return formatPercentage(dataRow.frequency, 0);
+    }
+
+    if (column.key === 'action') {
+      return (
+        <Link
+          to=""
```
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:251]: No idea what these will be yet, they don't quite match the mocks anyway, I'll remove it for now anyway

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    const currentSortKind = currentSort ? currentSort.kind : undefined;
+
+    return (
+      <SortLink
+        align="left"
+        title={columnInfo.name}
+        direction={currentSortKind}
+        canSort={canSort}
+        generateSortLink={generateSortLink}
+        onClick={() => {}}
```
[the remark this answers is not in the document]
Comment (k-fish) [reply static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:154]: Yep.

**Review by Zylphrex — APPROVED** [review]
The UI needs a few more iterations but LGTM 👍

**Review by dashed — APPROVED** [review]

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+    if (column.key === 'action') {
+      return (
+        <Link
+          to=""
+          onClick={() =>
+            this.handleTagValueClick(location, dataRow.tags_key, dataRow.tags_value)
+          }
+        >
+          <LinkContainer>
+            <IconAdd isCircled /> {t('Add to filter')}
```
Comment (dashed) [inline static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:249]: Feedback: This link/action should be disabled once clicked. Adding additional duplicated search conditions won't be helpful.

File: static/app/views/performance/transactionSummary/transactionTags/content.tsx [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+  }
+
+  const tagOptions = getTagKeyOptions(tableData);
+
+  const defaultTag = tagOptions.suspectTags.length
+    ? tagOptions.suspectTags[0]
+    : tagOptions.otherTags.length
+    ? tagOptions.otherTags[0]
+    : '';
+  const [tagSelected, changeTagSelected] = useState(defaultTag);
```
Comment (dashed) [inline static/app/views/performance/transactionSummary/transactionTags/content.tsx:112]: It'll be cool if we lift this up to the URL in a follow up PR.

File: static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx (hunk 6 above)
Comment (k-fish) [reply static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx:249]: That's a good call.

**Pull request merged**